galley/pkg/settings/settings.proto

```diff
@@ -0,0 +1,226 @@
+    // Defaults to  "istio-system".
+    string deployment_namespace = 5;
+
+    // The name of the validation deployment. This, along with deployment_namespace, is used to set the
+    // ownerReference in the validatingwebhookconfiguration. This enables k8s to clean-up the cluster-scoped
+    // validatingwebhookconfiguration when the deployment is deleted.
+    // Defaults to "istio-galley"
+    string deployment_name = 6;
+
+    // The name of the k8s service of the validation webhook. This is used to verify endpoint readiness before
+    // registering the validatingwebhookconfiguration.
+    // Defaults to "istio-galley"
+    string service_name  = 7;
+
+    // TLS settings for validation webhook service.
+    TLS tls = 8;
+}
+
+// Config processing related settings
+message Processing {
+    // Settings for the upstream configuration source
+    Source source = 2;
+
+    // Settings for the MCP Service
+    Server server = 3;
+
+    // Configuration sinks that Galley should connect to and deliver configuration
+    repeated MCPService sinks = 4;
+
+    // DNS Domain suffix to use while constructing Ingress based resources.
+    // Defaults to "cluster.local"
+    string domain_suffix = 5;
+
+}
+
+message Source {
+    oneof source {
+        KubernetesSource kubernetes = 1;
+        FilesystemSource filesystem = 2;
+        MCPService mcp = 3;
+    }
+}
+
+message Server {
+    // Disable the server
+    // Defaults to false
+    bool disable = 1;
+
+    // Address to use for Galley's gRPC API, e.g. tcp://127.0.0.1:9092 or unix:///path/to/file
+    // Defaults to tcp://0.0.0.0:9901.
+    string address = 2;
+
+    // Maximum size of individual gRPC messages
+    // Defaults to 1024 * 1024.
+    int32 max_received_message_size = 3;
+
+    // Maximum number of outstanding RPCs per connection
+    // Defaults to 1024.
+    int32 max_concurrent_streams = 4;
+
+    // Use insecure gRPC communication
+    // Defaults to false.
+    bool insecure = 5;
+
+    // Optional whitelisting file for SPIFFE IDs. If the file is not found, then all incoming connections are
+    // allowed.
+    // Defaults to fail open.
+    string access_list_file = 6;
+
+    // Disable resource readiness checks. This allows Galley to start if not all resource types are supported
+    // Defaults to false
+    bool disable_resource_ready_check = 7;
+}
+
+message KubernetesSource {
+    // Resync period for rescanning Kubernetes resources
+    google.protobuf.Duration resync_period = 1;
+}
+
+message FilesystemSource {
+    // Path to a folder that contains file-based configuration for Istio.
+    // Required
+    string path = 1;
+}
+
+message MCPService {
+    // The address of the MCP Service to dial out to.
+    // Required
+    string address = 1;
+
+    // Use insecure gRPC communication
+    // Defaults to false.
+    bool insecure = 2;
+
+    // The client metadata to supply when connecting to this MCP Service.
+    map<string, string> client_metadata = 3;
```

Comment:
nit: s/metadata/annotations :)

Reply:
Do we want gRPC stream metadata and  resource annotations?